ex1/ex1.c

```diff
@@ -1,14 +1,32 @@
 // Write a program that calls `fork()`. Before calling `fork()`, have the main process access a variable
 // (e.g., x) and set its value to something (e.g., 100). What value is the variable in the child process?
 // What happens to the variable when both the child and parent change the value of x?
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
-int main(void)
+int main(int argc, char *argv[])
 {
-    // Your code here
+    printf("hello world (pid: %d)\n", (int) getpid());
+    int x;
+    x = 100;
+    int rc = fork();    
+    printf("%d", x);
+    // ------------------------------------------------ child process starts executing here
+    if (rc < 0) {    // fork failed; exit
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+    } else if (rc == 0) {    // child process satisfies this branch
+        printf("hello, child here (pid: %d) \n", (int) getpid());
+        x = 105;
+        printf("%d %s", x, "child");
+
+        
```

Comment:
Nice! I like that you are commenting your code, your future employers will appreciate this!